geoportailv3/views/preparelegend.py

```diff
@@ -0,0 +1,39 @@
+from pyramid.view import view_config
+from pyramid.response import Response
+from pyramid.httpexceptions import HTTPBadRequest
+import logging
+import urllib2
+from urlparse import urljoin
+import html5lib
+
+log = logging.getLogger(__name__)
+
+
+class LegendPreparer(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(route_name='preparelegend')
+    def preparelegend(self):
```

Comment:
Note that you don't need a class in this case, you could just use a function:

```py
@view_config(route_name='preparelegend')
def preparelegend(request):
    # …
    return Response(…)
```

Just a comment, feel free to use a class if you prefer.